Content/Items/Accessories/BloodManaRing.cs

```diff
@@ -9,60 +10,54 @@ public class BloodManaRing : ModItem
-            item.rare = ItemRarityID.Orange;
-            item.value = Item.buyPrice(gold: 1);
+            Item.width = 20;
+            Item.height = 26;
+            Item.accessory = true;
+            Item.consumable = true;
+            Item.rare = ItemRarityID.Orange;
+            Item.value = Item.buyPrice(gold: 1);
         }
 
         public override void UpdateAccessory(Terraria.Player player, bool hideVisual)
         {
-            player.magicDamage += 0.70f;
+            player.GetDamage(DamageClass.Magic) += 0.70f;
             player.lifeRegen += 8;
             player.GetModPlayer<OvermorrowModPlayer>().Bloodmana = true;
-            player.allDamage -= 0.50f;
+            player.GetDamage(DamageClass.Generic) -= 0.50f;
             player.statManaMax = 200;
         }
 
         public override void AddRecipes()
         {
-            ModRecipe recipe1 = new ModRecipe(mod);
-            recipe1.AddIngredient(ItemID.BandofStarpower, 1);
-            recipe1.AddIngredient(ItemID.BandofRegeneration, 1);
-            recipe1.SetResult(this, 1);
-            recipe1.AddRecipe();
-
-            ModRecipe recipe2 = new ModRecipe(mod);
-            recipe2.AddIngredient(ItemID.PanicNecklace, 1);
-            recipe2.AddIngredient(ItemID.BandofRegeneration, 1);
-            recipe2.SetResult(this, 1);
-            recipe2.AddRecipe();
-
-            ModRecipe recipe3 = new ModRecipe(mod);
-            recipe3.AddIngredient(mod.ItemType("SoulFire"), 1);
-            recipe3.AddIngredient(mod.ItemType("DrippingFlesh"), 1);
-            recipe3.SetResult(this, 1);
-            recipe3.AddRecipe();
-
-
-            ModRecipe recipe4 = new ModRecipe(mod);
-            recipe4.AddIngredient(ItemID.ManaCrystal, 1);
-            recipe4.AddIngredient(ItemID.LifeCrystal, 1);
-            recipe4.SetResult(this, 1);
-            recipe4.AddRecipe();
-
-            ModRecipe recipe5 = new ModRecipe(mod);
-            recipe5.AddIngredient(mod.ItemType("BloodGem"), 1);
-            recipe5.AddIngredient(mod.ItemType("CrystalMana"), 1);
-            recipe5.SetResult(this, 1);
-            recipe5.AddRecipe();
+            CreateRecipe()
+                .AddIngredient(ItemID.BandofStarpower, 1)
+                .AddIngredient(ItemID.BandofRegeneration, 1)
+                .Register();
+
+            CreateRecipe()
+                .AddIngredient(ItemID.PanicNecklace, 1)
+                .AddIngredient(ItemID.BandofRegeneration, 1)
+                .Register();
+
+            CreateRecipe()
+                .AddIngredient<SoulFire>()
+                .AddIngredient<MutatedFlesh>()
+                .Register();
+
+            CreateRecipe()
+                .AddIngredient(ItemID.ManaCrystal)
+                .AddIngredient(ItemID.LifeCrystal)
+                .Register();
+
+            CreateRecipe()
+                .AddIngredient<BloodGem>()
+                .AddIngredient<CrystalMana>()
+                .Register();
```

Comment:
New recipe style. It's pretty much exactly the same, but more flexible and quite a bit more elegant.